pkg/prometheus/operator.go

```diff
@@ -72,6 +72,12 @@ type Operator struct {
 
 	statefulsetErrors prometheus.Counter
 
+	// triggerByCounter is a map of maps of Prometheus counters. The first map
+	// represents each Kubernetes object, the second map represents its actions
+	// (add, delete, update). These metrics help understand what triggers the
+	// Prometheus Operator to reconcile its created objects
+	triggerByCounter map[string]map[string]prometheus.Counter
```

Comment:
What you are doing here is equivalent to a `CounterVec`, we should use that instead.